kb(slider):add kb for max parameter zero in slider (#564)
telerik/blazor-docs: knowledge-base/slider-max-parameter-zero-error.md

---
title: Requires the Max parameter error in Slider

> description: Max parameter of zero throws an error

Reviewer: Blazor Slider requires a Max parameter and throws an error if zero is set.

type: troubleshooting
page_title: Requires the Max parameter error in Slider
slug: slider-kb-max-parameter-zero-error
position: 
tags: 
ticketid: 1541425
res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>Slider for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

> I'm using the Telerik Slider, and I noticed that I can't set the Max to 0. When I set the Max parameter to zero, the component throws an error.

Reviewer: "noticed" doesn't contribute to customer context, as a person will not search by this word.

Consider quoting the exact error message below too:

```suggestion
I'm using the Telerik Slider, and I can't set Max to 0. When I set the Max parameter to zero, the component throws an error.
```

## Error Message
>warning ArgumentException: Telerik.Blazor.Components.TelerikSlider`1[System.Int32] requires the Max parameter.

## Cause\Possible Cause(s)

> Since the max parameter is a required parameter, we are checking if the user has provided value. That being said, if the user doesn't set any value for the Max parameter, the parameter will stay with its default value and throw an error. The default value for a non-nullable type (like int) is 0, and when you manually set 0 to the Max parameter, the component acts like there is no Max parameter declared and throws an error again.

Reviewer: ```suggestion
The `Max` parameter is a required attribute and the component is checking if it is set. If the developer doesn't set any value for Max, the parameter value will remain the default value of the `int` type and throw an error. The default value for the non-nullable `int` type is 0, and when you explicitly set 0 to the Max parameter, the component acts like there is no Max parameter declared and still throws the error.
```

## Solution

> You can avoid this exception by using a nullable type of the bound value.

Reviewer: ```suggestion
You can avoid the exception by using a nullable type for the Slider `Value`.
```
